src/core/Jwt.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+class Jwt {
+  constructor (encodedJwt) {
+    this._encodedJwt = encodedJwt;
+
+    this._userId = null;
+    this._expiresAt = null;
+
+    this._decode();
+  }
+
+  get encodedJwt () {
+    return this._encodedJwt;
+  }
+
+  get userId () {
+    return this._userId;
+  }
+
+  get expiresAt () {
+    return this._expiresAt;
+  }
+
+  get expired () {
+    return Date.now() > this.expiresAt;
+  }
+
+  _decode () {
+    const 
+      [, payloadRaw, ] = this._encodedJwt.split('.'),
+      payload = JSON.parse(new Buffer(payloadRaw, 'base64').toString());
```

Comment:
you first need to check that `payloadRaw` is not undefined (invalid token) and you also need to add a `try..catch` around JSON.parse, in case the payload is not in JSON format (again, invalid token).